ipfs: remove version on go dependency

bpresles opened the pull request:
Makes ipfs formula buildable on Apple Silicon

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

SMillerDev commented:
The install block can also be massively simplified now:

```patch
  def install
-    ENV["GOPATH"] = buildpath
-    (buildpath/"src/github.com/ipfs/go-ipfs").install buildpath.children
-    cd("src/github.com/ipfs/go-ipfs") { system "make", "install" }
-    bin.install "bin/ipfs"
-    cd("src/github.com/ipfs/go-ipfs") { bash_completion.install "misc/completion/ipfs-completion.bash" }
+   system "make", "install"
+   bash_completion.install "misc/completion/ipfs-completion.bash"
  end
```

bpresles commented:
@SMillerDev Thanks, I modified the install block. Also I removed the version on "go" dependency because of check failure in "brew audit --strict ipfs".

I don't know if it safe though, it builds for now, but can it break if go gets upgraded ?

SMillerDev commented:
> I don't know if it safe though, it builds for now, but can it break if go gets upgraded ?

If the go formula gets updated all of its dependencies will be tested and if they break they'll be put back to an older go version.

bpresles commented:
@SMillerDev SMillerDev Hmm it built correctly and appeared to install, but with the new install instructions the binary is not installed. The install block instructions you provided doesn't seem to work. I'll revert for now.
